src/components/UI/Table/Table.vue

```diff
@@ -0,0 +1,199 @@
+                  :key="columnName"
+                  class="sm:px-6 sm:py-3 px-3 py-1.5 whitespace-normal"
+                  :class="{
+                    'hidden sm:table-cell': columnIndex != 0,
+                  }"
+                >
+                  <div class="flex">
+                    <div v-if="isComponent(entry[columnName])" class="">
+                      <PlioListItem :plioId="entry[columnName].value" :org="org">
+                      </PlioListItem>
+                    </div>
+
+                    <div v-else class="sm:text-lg text-sm font-medium text-gray-900">
+                      {{ entry[columnName] }}
+                    </div>
+                  </div>
+                </td>
+
+                <td
+                  class="sm:px-6 sm:py-3 px-3 py-1.5 whitespace-normal table-cell sm:hidden"
+                >
+                  <div class="flex flex-col">
+                    <div
+                      v-for="(columnName, columnIndex) in columns"
+                      :key="columnName"
+                      class="sm:text-lg text-sm font-medium text-gray-900 min-w-max"
+                      :class="{ hidden: columnIndex == 0 }"
+                    >
+                      {{ capitalize(columnName) }} : {{ entry[columnName] }}
+                    </div>
+                  </div>
+                </td>
+              </tr>
+            </tbody>
+          </table>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import PlioListItem from "@/components/UI/ListItems/PlioListItem.vue";
+import IconButton from "@/components/UI/Buttons/IconButton.vue";
+
+export default {
+  props: {
+    data: Array,
+    columns: Array,
+    org: {
+      default: "",
+      type: String,
+    },
+  },
+  components: {
+    PlioListItem,
+    IconButton,
+  },
+  data() {
+    return {
+      sortKey: "",
+      filterKey: "",
+      sortOrders: {},
+      searchIconConfig: {
+        enabled: true,
+        iconName: "search-solid",
+        iconClass: "text-yellow-600 h-5 w-5",
+      },
+      searchButtonClass: "",
+    };
+  },
+
+  created() {
+    this.initialiseSortOrders();
+  },
+
+  computed: {
+    filteredData() {
+      // TODO - sorting and filtering
+
+      // const sortKey = this.sortKey;
+      // const filterKey = this.filterKey && this.filterKey.toLowerCase();
+      // const order = this.sortOrders[sortKey] || 1;
+      let data = this.data;
+      // if (filterKey) {
+      //   data = data.filter(function (row) {
+      //     return Object.keys(row).some(function (key) {
+      //       return String(row[key]).toLowerCase().indexOf(filterKey) > -1;
+      //     });
+      //   });
+      // }
+      // if (sortKey) {
+      //   data = data.slice().sort(function (a, b) {
+      //     a = a[sortKey];
+      //     b = b[sortKey];
+      //     return (a === b ? 0 : a > b ? 1 : -1) * order;
+      //   });
+      // }
```

Comment:
are these needed for later?

Reply:
yes, to implement the sorting and searching logic